stdlib/private/StdlibUnittest/StdlibUnittest.swift

```diff
@@ -125,16 +123,16 @@ public func expectFailure(
   stackTrace: SourceLocStack = SourceLocStack(),
   showFrame: Bool = true,
   file: String = #file, line: UInt = #line, invoking body: () -> Void) {
-  let startAnyExpectFailed = _anyExpectFailed
-  _anyExpectFailed = false
+  let startAnyExpectFailed = _anyExpectFailed.load()
+  _anyExpectFailed.store(0) /*false*/
```

Comment:
This `expectFailure` function is fundamentally unsafe if multiple threads are calling it at the same time. 

Luckily we don't use `expectFailure` much outside of StdlibUnittest's own tests, so it's probably not an issue -- but it's definitely worth adding a FIXME for it.

Reply:
Sure we can add a FIXME, just not sure I understand why it is broken, maybe I'm missing something 😅 Can you point out why?

Reply:
The issue is that there is only a single, global `_anyExpectFailed` flag; if two or more threads run `expectFailure` at the same time, they can clobber each other's state.

However, on second thought, this isn't really a practical issue, because the entire StdlibUnittest framework only supports sequential testing -- two or more tests can never run at the same time in the same process. So we can leave this as is; there is no need to add a FIXME here, since it would apply to the entire module anyway.